### ISSUE TITLE
when you construct a concept graph, are you also use dialogues in valid or testset in Personachat?

### ISSUE THREAD

**Issue by kimwongyuda** [RowitZou/CG-nAR#1]
when you construct a concept graph, are you also use dialogues in valid or testset in Personachat?

**Comment by RowitZou** [comment]
Since the graph construction strategy is not our main focus, it was largely inspired by previous works (Zhou et al., 2018 and Xu et al., 2020, as cited in our paper), which generally associate the graph with the whole corpus (including valid/test sets). For example, in Zhou et al. 2018, if a post-response pair can not be connected by any triple in ConceptNet (that is, one entity appears in the post and the other in the response), the pair will be removed. 

The constructed concept graph contains all candidate target concepts, and the model has no opportunity to know which concept is from the valid/test sets. Additionally, for a fair comparison, we apply the same concept graph to all those methods with concept planning, avoiding the influence of graph construction strategy.

**Comment by kimwongyuda** [comment]
Thank you for replying!

I just wonder your opinion about below situation.

Suppose that (job, doctor) pair is from the testset and the model can benefit from that pair during training. The fact that the information from the testset being used during training seems to make the training process unfair even though the model doesn't know that the pair is from the testset.

**Comment by RowitZou** [comment]
We understand your concern about the potential label leaking, and it might be unfair to those baselines that do not employ graph information, e.g., vanilla transformer.  However, most research works about knowledge-based dialogue generation have made the strong assumptions that the training set and evaluation set are i.i.d., and a high-quality and powerful knowledge base is available (unfortunately such a knowledge base is hard to construct in real situations). The ideal knowledge base can generally cover the information of valid/test set, facilitating both the process of training and inference.

**Comment by kimwongyuda** [comment]
thank you for replying!
it helps me a lot.
